### PR TITLE
Add a Jenkinsfile

### PR DESCRIPTION
**Pull request by aheritier**
Using pipeline-model-definition-plugin
https://github.com/jenkinsci/pipeline-model-definition-plugin/wiki/Getting-Started

### CODE REVIEW

**Comment by aheritier** [comment]
Please @jenkinsci/code-reviewers @reviewbybees cc @abayer @olamy 
What is the process @daniel-beck @rtyler to host a build on the new CI infra ?

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    stages {
+        stage("Build") {
+            sh 'mvn package -B -Dmaven.test.failure.ignore=true'
+        }
+    }
+
+    postBuild {
+        success {
+            archive "**/target/**/*.jar,**/target/**/*.hpi"
+            junit '**/target/surefire-reports/*.xml'
```
Comment (abayer) [inline Jenkinsfile]: I think you will want this for unstable too.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+pipeline {
+    agent label:"docker", docker: "maven:3.3.9-jdk-8"
```
Comment (olivergondza) [inline Jenkinsfile]: Does it require node with docker installed?

File: Jenkinsfile (hunk 1 above)
Comment (abayer) [inline Jenkinsfile]: Check that, I'm dumb. Forgot that `unstable` won't get set until we do the `junit` anyway! I could see an argument for still doing the junit reports even if the build failed past tests - i.e., findbugs failures or the like - but that's not a big deal.

**Comment by abayer** [comment]
:bee:

File: Jenkinsfile (hunk 2 above)
Comment (jglick) [inline Jenkinsfile]: Yes.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    environment {
+      MAVEN_OPTS = "-Xmx1024m"
+    }
+
+    stages {
+        stage("Build") {
+            sh 'mvn package -B -Dmaven.test.failure.ignore=true'
```
Comment (jglick) [inline Jenkinsfile]: BTW `=true` can be elided.

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        }
+    }
+
+    postBuild {
+        success {
+            archive "**/target/**/*.jar,**/target/**/*.hpi"
```
Comment (jglick) [inline Jenkinsfile]: FWIW I recommend using `'` when you are not doing string interpolation.

**Comment by abayer** [comment]
Basically just ask me nicely and I'll set it up. =)

**Comment by rtyler** [comment]
@abayer asking nicely is not sufficient IMHO. I think emailing the infrastructure list or file an INFRA is more acceptable than asking a single person.

There are still Pipeline-related resource allocation/timeout issues (some which are awaiting release) which prevent broader roll-out of plugin builds on ci.jenkins.io. So I would prefer to keep this as ticketed-opt-in instead of somebody with admin going into org folders and editing whitelists without broader record of change.

**Comment by abayer** [comment]
Okiedokie!

**Comment by aheritier** [comment]
We will redo it now the DSL is stable

**Comment by jglick** [comment]
I think you would rather want to use the standard

```groovy
buildPlugin()
```

**Comment by aheritier** [comment]
yes @jglick 
Now it exists this PR makes no sense

**Pull request closed without merging**